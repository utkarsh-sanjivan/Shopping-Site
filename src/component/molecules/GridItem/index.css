.grid-container {
    width: 300px;
    padding: 10px;
    /* border: 1px solid   #aaa; */
    display: flex;
    flex-direction: column;
    align-items: center;
}

.grid-container:hover {
    border: 1px solid #dedede;
    padding-left: 12px;
    box-shadow: 8px 5px 7px #000;
}

.product-image {

}

.product-text {
    width: 100%;
    text-align: center;
    font-size: 20px;
    line-height: 40px;
    font-weight: 600;
}

.product-description {
    width: 100%;
    text-align: justify;
    line-height: 20px;
    padding: 0 5px;
}

.product-info {
    width: 100%;
    text-align: justify;
    line-height: 20px;
    margin-bottom: 10px;
}