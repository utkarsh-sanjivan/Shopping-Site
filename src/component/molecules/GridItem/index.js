import React from 'react';
import './index.css';

export default function GridItem(props) {
    return (
        <div className='grid-container'>
            <img className='product-image' src={props.image} alt={'Product'} width={300}/>
            <div className='product-text'>{props.name}</div>
            <div className='product-description'>{props.description}</div>
            <div className='product-info'><b>Price:</b> {props.price}</div>
            <div className='product-info'><b>Quality:</b> {props.quantity}</div>
            {parseInt(props.quantity)>0?<div  className='product-button' onClick={() => props.handleSelect(props.id)}>Add to Cart</div>: null}
        </div>
    );
}