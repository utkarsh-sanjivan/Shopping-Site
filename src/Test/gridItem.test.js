import React from 'react';
import { create } from "react-test-renderer";
import GridItem from '../component/molecules/GridItem/index';

let wrapper;
let root;
const renderComponent = props => <GridItem {...props} />;
const baseProps = {
    key: '1',
    image: '',
    name: 'Product1',
    description: 'lorem ipsum',
    price: '100.00',
    quantity: '0',
    handleSelect: () => {},
};

beforeEach(() => {
  wrapper = create(renderComponent(baseProps));
  root=wrapper.root;
});

// Unit Test Cases for testing Grid Item Component with 0 quantity.
describe('Testing Grid Item Component with 0 quantity', () => {
    // Test for matching the snapshot when there is 0 quantity.
    it('Grid Item Component should render properly', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    // Test for numbers of div DOM element in the Grid Item component when there is 0 quantity.
    it('should atleast 5 div component', () => {
        expect(root.findAllByType('div').length).toEqual(5);
    });

    // Test for numbers of img DOM element in the Grid Item component when there is 0 quantity.
    it('should atleast 1 img component', () => {
        expect(root.findAllByType('img').length).toEqual(1);
    });
});

describe('Testing Grid Item Component with 4 quantity', () => {
    // Test for matching the snapshot when there is 4 quantity.
  it('Grid Item Component should render properly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

    // Test for numbers of div DOM element in the Grid Item component when there is 4 quantity.
    it('should atleast 5 div component', () => {
        expect(root.findAllByType('div').length).toEqual(5);
    });

    // Test for numbers of div DOM element in the Grid Item component when there is 4 quantity.
    it('should atleast 1 img component', () => {
        expect(root.findAllByType('img').length).toEqual(1);
    });
});
