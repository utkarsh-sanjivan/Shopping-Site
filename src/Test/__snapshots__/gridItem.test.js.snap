// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Grid Item Component Rendering Grid Item Component should render properly 1`] = `
<div
  className="grid-container"
>
  <img
    alt="Product"
    className="product-image"
    src=""
    width={300}
  />
  <div
    className="product-text"
  >
    Product1
  </div>
  <div
    className="product-description"
  >
    lorem ipsum
  </div>
  <div
    className="product-info"
  >
    <b>
      Price:
    </b>
     
    100.00
  </div>
  <div
    className="product-info"
  >
    <b>
      Quality:
    </b>
     
    10
  </div>
  <div
    className="product-button"
    onClick={[Function]}
  >
    Add to Cart
  </div>
</div>
`;
